monai/transforms/post/array.py

```diff
@@ -858,3 +871,97 @@ def __call__(self, image: NdarrayOrTensor) -> torch.Tensor:
         grad = torch.cat([grad_h, grad_v], dim=1)
         grad, *_ = convert_to_dst_type(grad.squeeze(0), image_tensor)
         return grad
+
+
+class GetInstanceLevelSegMap(Transform):
+    """
+    Process Nuclei Prediction with XY Coordinate Map.
+
+    Args:
+        threshold_pred: threshold the float values of prediction to int 0 or 1 with specified theashold. Defaults to 0.5.
+        threshold_overall: threshold the float values of overall gradient map to int 0 or 1 with specified theashold.
+            Defaults to 0.4.
+        min_size: objects smaller than this size are removed. Defaults to 10.
+        sigma: std. could be a single value, or `spatial_dims` number of values. Defaults to 0.4.
+        kernel_size: the size of the Sobel kernel. Defaults to 17.
+        radius: the radius of the disk-shaped footprint. Defaults to 2.
+
+    """
+
+    backend = [TransformBackends.TORCH]
+
+    def __init__(
+        self,
+        threshold_pred: float = 0.5,
+        threshold_overall: float = 0.4,
+        min_size: int = 10,
+        sigma: Union[Sequence[float], float, Sequence[torch.Tensor], torch.Tensor] = 0.4,
+        kernel_size: int = 17,
+        radius: int = 2,
+    ) -> None:
+        self.sigma = sigma
+        self.radius = radius
+
+        self.sobel_gradient = SobelGradients(kernel_size=kernel_size)
+        self.remove_small_objects = RemoveSmallObjects(min_size=min_size)
+        self.pred_discreter = AsDiscrete(threshold=threshold_pred)
+        self.overall_discreter = AsDiscrete(threshold=threshold_overall)
+        self.fill_holes = FillHoles()
+
+    def __call__(self, pred: NdarrayOrTensor, hover: NdarrayOrTensor) -> torch.Tensor:  # type: ignore
+        """
+        Args:
+            pred: the probability map output of the NP branch, shape must be [1, H, W, [D]].
+            hover: the horizontal and vertical distances of nuclear pixels to their centres of mass output from the HV branch,
+                shape must be [2, H, W, [D]].
+
+        Returns:
+            instance labelled segmentation map with shape [1, H, W, [D]].
+        """
+        pred_t = convert_to_tensor(pred, track_meta=get_track_meta())
+        hover_t = convert_to_tensor(hover, track_meta=get_track_meta())
+
+        hover_h = hover_t[0:1, ...]
+        hover_v = hover_t[1:2, ...]
+
+        # processing
+        blb = self.pred_discreter(pred_t)
```

Comment:
Here also we can keep the name of `pred`.

Reply:
Updated it! thanks!